addons/phone_validation/tests/test_phonenumbers_patch.py

```diff
@@ -75,3 +75,20 @@ def test_region_PA_monkey_patch(self):
         parsed = phonenumbers.parse('+507 833 8744')
         self.assertTrue(phonenumbers.is_valid_number(parsed))
         self.assertEqual(parsed.national_number, 8338744)
+
+    def test_region_SN_monkey_patch(self):
+        """Makes sure that patch for Senegalese phone numbers work"""
+        if not phonenumbers:
+            self.skipTest('Cannot test without phonenumbers module installed.')
+        version_to_patch_up_to = '8.12.29'
+
+        # Check if the patch was applied to appropriate versions of phonenumber module
+        if parse_version(phonenumbers.__version__) < parse_version(version_to_patch_up_to):
+            self.assertEqual(phonenumbers.PhoneMetadata._region_available['SN'], phonenumbers_patch._local_load_region,
+                f"The phonenumbers module should have been patched up to version {version_to_patch_up_to}")
+            phone_number = phonenumbers.parse("+221750142092", region="SN", keep_raw_input=True)
+            self.assertTrue(phonenumbers.is_valid_number(phone_number),
+                "The number +221750142092 is not valid (it should be). phonenumbers patch probably was not applied")
```

Comment:
Isn't this assert valid for new versions also ^^ ? In the end what we want is to correctly parse numbers, isn't it ?

Reply:
Up for this remark: test should always be valid, isn't it ?

Reply:
absolutely yes, we should first and foremost check that the number that we consider correct is parsed fine. Whether it is because of monkey patch is secondary, adding that to the fix.